### PR TITLE
[Intel GPU][pt2e]: Collapse 3D input to 2D for matmul in qlinear_pointwise_binary fusion

### PR DESCRIPTION
**Pull request by ZhiweiYan-96**
# Motivation
During the `qlinear_pointwise_binary` lowering pass, dim collapsing only occurs when post-ops is `add`. It is the responsibility of  C++ kernels to handle dimension for post-ops `sum`

# Details
This PR explicitly reshape input from 3D to 2D in op `qlinear_pointwise_binary`. Besides, we refractor implementation `qlinear_pointwise_binary.tensor` to call `qlinear_pointwise_binary` for removing duplicated codes.

# UT testing
`python test/inductor/test_mkldnn_pattern_matcher.py -k test_qlienar_add_xpu`

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #148522
* __->__ #148423

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

### CODE REVIEW

File: test/xpu/test_xpu_inductor_quantizer.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            relative_err = torch.mean(torch.abs(res - ref) / ref.abs().clamp(1e-6))
+            self.assertTrue(relative_err < 0.1)
+            matcher_check_fn()
+
+    def test_qlinear_pointwise_binary_3d(self):
+        class Model(torch.nn.Module):
+            def __init__(self):
+                super(Model, self).__init__()
+                self.linear = torch.nn.Linear(10, 10)
+                self.relu = torch.nn.ReLU()
```
Comment (liangan1) [inline test/xpu/test_xpu_inductor_quantizer.py:73]:
```suggestion

```

File: test/xpu/test_xpu_inductor_quantizer.py
Comment (EikanWang) [inline test/xpu/test_xpu_inductor_quantizer.py:1]: @ZhiweiYan-96 , why do we need to add a dedicated test file? I suppose it should reuse other test files. Right?

**Review by EikanWang — APPROVED** [review]

**Comment by ZhiweiYan-96** [comment]
Hi @jansel @desertfire,
Would you mind reviewing the code changes on the inductor side? Great appreciation for your suggestions.. Please note that we've already conducted an internal review of the changes in the XPU backend (aten/src/ATen/native/mkldnn/xpu/), so feel free to focus on the new additions, or review the entire change as you prefer. 
Thanks again for your time.

**Review by jansel — APPROVED** [review]
